src/itest/groovy/com/hierynomus/sshj/signature/KeyWithCertificateSpec.groovy

```diff
@@ -62,4 +73,43 @@ class KeyWithCertificateSpec extends IntegrationBaseSpec {
                 "id_ed25519_384_rfc4716_signed_by_rsa",
         ]
     }
+
+    @Unroll
+    def "accepting a signed host public key with type #hostKeyAlgo"() {
+        given:
+        File knownHosts = Files.createTempFile("known_hosts", "").toFile()
+        knownHosts.deleteOnExit()
+
+        and:
+        File caPubKey = new File("src/itest/resources/keyfiles/certificates/CA_rsa.pem.pub")
+        String knownHostsFileContents = "" +
+                "@cert-authority $SERVER_IP ${caPubKey.text}" +
+                "\n@cert-authority [$SERVER_IP]:$DOCKER_PORT ${caPubKey.text}"
```

Comment:
By the way, OpenSSH would require only the first line. It always checks for a host entry without a specified port when fails to find a matching record with a port.
This pull request is already quite big, so I'll address this issue later.